crashtracker/src/collector/crash_handler.rs

```diff
@@ -324,75 +436,88 @@ fn handle_posix_signal_impl(signum: i32, sig_info: *mut siginfo_t) -> anyhow::Re
     anyhow::ensure!(!metadata_ptr.is_null(), "No crashtracking metadata");
     let (_metadata, metadata_string) = unsafe { metadata_ptr.as_ref().context("metadata ptr")? };
 
+    let receiver_config = RECEIVER_CONFIG.swap(ptr::null_mut(), SeqCst);
+    anyhow::ensure!(
+        !receiver_config.is_null(),
+        "No crashtracking receiver config"
+    );
+    let receiver_config = unsafe { receiver_config.as_ref().context("receiver config")? };
+
+    // Since we've gotten this far, we're going to start working on the crash report. This
+    // operation needs to be mindful of the total walltime elapsed during handling. This isn't only
+    // to prevent hanging, but also because services capable of restarting after a crash experience
+    // crashes as probabalistic queue-holding events, and so crash handling represents dead time
+    // which makes the overall service increasingly incompetent at handling load.
+    let timeout_ms = config.timeout_ms;
+    let start_time = Instant::now(); // This is the time at which the signal was received
```

Comment:
is this signal safe?

Reply:
Good question.  It is, but more importantly this should be asserted and documented more carefully.  I've added a section at the top of the file that indicates which of the syscalls we use are async-signal safe and provides a link.

Of course, we do use some convenience wrappers through `nix`, so if `nix` calls free on something then that might be problematic.  I hope we can isolate and remove these over time.